pennylane/measurements/probs.py

```diff
@@ -299,23 +300,24 @@ def marginal_prob(self, prob, wire_order, batch_size):
         # reshape the probability so that each axis corresponds to a wire
         num_device_wires = len(wire_order)
         shape = [2] * num_device_wires
+        desired_axes = np.argsort(np.argsort(mapped_wires))
+        flat_shape = (-1,)
         if batch_size is not None:
             shape.insert(0, batch_size)
             inactive_wires = [idx + 1 for idx in inactive_wires]
+            desired_axes = np.insert(desired_axes + 1, 0, 0)
+            flat_shape = (batch_size, -1)
+
         # prob now is reshaped to have self.num_wires+1 axes in the case of broadcasting
         prob = qml.math.reshape(prob, shape)
 
         # sum over all inactive wires
-        flat_shape = (-1,) if batch_size is None else (batch_size, -1)
-        prob = qml.math.reshape(qml.math.sum(prob, axis=tuple(inactive_wires)), flat_shape)
+        if inactive_wires:
+            prob = qml.math.sum(prob, axis=tuple(inactive_wires))
```

Comment:
What is the difference with the `reduced_sum` used before?

Reply:
this was actually just using `qml.math.sum` before as well, but yeah the old devices used `reduce_sum`. As you can see on the _qubit_device [here](https://github.com/PennyLaneAI/pennylane/blob/1774fe36f31f7de6266cf44983c85dfb561ec0fc/pennylane/_qubit_device.py#L113), it was just `np.sum` with axis provided. So we do the same thing here!